cirq-core/cirq/sim/state_vector_simulation_state.py

```diff
@@ -226,7 +226,7 @@ def prepare_into_buffer(k: int):
         ]
         p = prng.random()
         weight = None
-        fallback_weight = 0
+        fallback_weight = np.float64(0)
```

Comment:
```suggestion
        fallback_weight = 0.0
```

Reply:
I get 

`cirq-core/cirq/sim/state_vector_simulation_state.py:228: error: Incompatible types in assignment (expression has type "float", variable has type "floating[Any]")  [assignment]
` when we do this. I can either leave this as-is or do `0.0` here and ` weight = float(np.linalg.norm(self._buffer) ** 2)` a little bit more down.